datalake-test-utils/src/main/scala/bio/ferlab/datalake/testutils/ClassGenerator.scala

```diff
@@ -71,8 +71,13 @@ object ClassGenerator {
         case structField: StructField => structField.toString()
       }
     } else {
-      val countNulls: Column = df.schema.fieldNames.map(c => functions.when(col(c).isNull, 1).otherwise(0)).reduce((a, b) => a + b)
-      val values: Row = df.withColumn("countNulls", countNulls).orderBy(asc("countNulls")).drop("countNulls").head()
+      val values: Row = (
+        if (df.count() == 1) {
+          df.head()
+        } else {
+          val countNulls: Column = df.schema.fieldNames.map(c => functions.when(col(c).isNull, 1).otherwise(0)).reduce((a, b) => a + b)
```

Comment:
the following code is reaching stackoverflow when the DF fields stuct is enormous, that was my case with GnomadV4SV which got 400+ fields

Taking first line if the test file got only one.